### ISSUE TITLE
Support 'object' schema fields

### ISSUE THREAD

**Issue by Andrew-Crosby** [autotraderuk/fastapi-mlflow#11]
Models returning a string field (e.g. classification models) may have the following output schema in mlflow
```
Tensor (dtype: object, shape: [-1])
```
We should map 'object' schema types to Python 'string' types:
https://github.com/autotraderuk/fastapi-mlflow/blob/98701a0f5af535ceedda78c07759f5a3b6b62b8e/fastapi_mlflow/_mlflow_types.py#L10-L23

**Comment by bloomonkey** [comment]
Thanks for reporting @Andrew-Crosby

Do you have an example of how such an mlflow schema could be created? I would have hoped that [this test](https://github.com/autotraderuk/fastapi-mlflow/blob/main/tests/conftest.py#L51) would have covered string/object types, but maybe this bug only manifests when the return type is an array/tensor?

**Comment by Andrew-Crosby** [comment]
Scikit-learn classifiers produce an output array of type 'object'

This example code
```
X = pd.DataFrame({"feature": [1,2]})
y = pd.Series(["yes", "no"])

preds = LogisticRegression().fit(X, y).predict(X)

mlflow.models.infer_signature(X, preds)
```
produces the following signature:
```
inputs: 
  ['feature': long]
outputs: 
  [Tensor('object', (-1,))]
```

**Comment by bloomonkey** [comment]
I'm having a hard time reproducing this without introducing `sklearn` as a dependency. i.e. adding the following model to the test suite continues to pass all tests

```
class StrModel(PythonModel):
    def predict(
        self, context: PythonModelContext, model_input: pd.DataFrame
    ) -> npt.ArrayLike:
        return np.array(["42"] * len(model_input))
```

**Comment by bloomonkey** [comment]
Fixed. Will release along with a few other fixes as 0.4.1 within the next week
